plutus-contract/src/Plutus/Contract/Wallet.hs

```diff
@@ -253,19 +251,20 @@ export
     :: P.Params
     -> UnbalancedTx
     -> Either CardanoLedgerError ExportTx
-export params utx =
-    let requiredSigners = Set.toList (unBalancedTxRequiredSignatories utx)
-        fromCardanoTx ctx = do
-            utxo <- fromPlutusIndex $ P.UtxoIndex (unBalancedTxUtxoIndex utx)
+export params (UnbalancedEmulatorTx tx sigs utxos) =
+    let requiredSigners = Set.toList sigs
+     in ExportTx
+        <$> bimap Right (C.makeSignedTransaction []) (CardanoAPI.toCardanoTxBody params requiredSigners tx)
+        <*> first Right (mkInputs (P.pNetworkId params) utxos)
+        <*> pure (mkRedeemers tx)
+export params (UnbalancedCardanoTx tx utxos) =
+    let fromCardanoTx ctx = do
+            utxo <- fromPlutusIndex $ P.UtxoIndex utxos
             makeTransactionBody params utxo ctx
      in ExportTx
-        <$> fmap (C.makeSignedTransaction [])
-                 (either
-                     fromCardanoTx
-                     (first Right . CardanoAPI.toCardanoTxBody params requiredSigners)
-                     (unBalancedTxTx utx))
-        <*> first Right (mkInputs (P.pNetworkId params) (unBalancedTxUtxoIndex utx))
-        <*> either (const $ Right []) (Right . mkRedeemers) (unBalancedTxTx utx)
+        <$> fmap (C.makeSignedTransaction []) (fromCardanoTx tx)
+        <*> first Right (mkInputs (P.pNetworkId params) utxos)
+        <*> pure []
```

Comment:
I know it's not part of this ticket, but could you refactor `mkRedeemers` here using `getCardanoTxRedeemers`. Otherwise I'm pretty sure we will forget.

Reply:
mhhh... `getCardanoTxRedeemers` is for `CardanoTx` while we're dealing with a plutus `Tx` here, so I don't think we can make it.

Reply:
Ok, then at least add a TODO.